vcd/resource_vcd_access_control_vapp_test.go

```diff
@@ -0,0 +1,337 @@
+					testAccFindValuesInSet(resourceAC2, "shared", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessFullControl,
+					}),
+
+					resource.TestCheckResourceAttr(resourceAC2, "shared_to_everyone", "false"),
+					resource.TestCheckResourceAttr(resourceAC2, "shared.#", "2"),
+					testAccFindValuesInSet(resourceAC2, "shared", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessFullControl,
+					}),
+					testAccFindValuesInSet(resourceAC2, "shared", map[string]string{
+						"subject_name": "ac-user2",
+						"access_level": types.ControlAccessReadWrite,
+					}),
+
+					resource.TestCheckResourceAttr(resourceAC3, "shared_to_everyone", "false"),
+					resource.TestCheckResourceAttr(resourceAC3, "shared.#", "3"),
+					testAccFindValuesInSet(resourceAC3, "shared", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessFullControl,
+					}),
+					testAccFindValuesInSet(resourceAC3, "shared", map[string]string{
+						"subject_name": "ac-user2",
+						"access_level": types.ControlAccessReadWrite,
+					}),
+					testAccFindValuesInSet(resourceAC3, "shared", map[string]string{
+						"subject_name": "ac-user3",
+						"access_level": types.ControlAccessReadOnly,
+					}),
+				),
+			},
+			resource.TestStep{
+				Config: updateText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdVappAccessControlExists(resourceAC0, testConfig.VCD.Org, testConfig.VCD.Vdc),
+					resource.TestCheckResourceAttr(resourceAC0, "shared_to_everyone", "false"),
+					resource.TestCheckResourceAttr(resourceAC1, "shared.#", "1"),
+					testAccFindValuesInSet(resourceAC1, "shared", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessReadWrite,
+					}),
+					resource.TestCheckResourceAttr(resourceAC2, "shared.#", "2"),
+					testAccFindValuesInSet(resourceAC2, "shared", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessReadWrite,
+					}),
+					resource.TestCheckResourceAttr(resourceAC3, "shared.#", "3"),
+					testAccFindValuesInSet(resourceAC3, "shared", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessReadWrite,
+					}),
+				)},
+
+			// Tests import by name
+			resource.TestStep{
+				Config:            configText,
+				ResourceName:      "vcd_access_control_vapp.AC-Vapp1",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdOrgVdcObject(testConfig, "Vapp-AC-1"),
+			},
+			// Tests import by ID
+			resource.TestStep{
+				Config:            configText,
+				ResourceName:      "vcd_access_control_vapp.AC-Vapp2",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdViaResource("vcd_access_control_vapp.AC-Vapp2"),
+			},
+		},
+	})
+}
+
```

Comment:
Would be good test could verify that delete also works

Reply:
Delete works implicitly. The resource gets removed, and we check that it is not there anymore